### PR TITLE
Fix the TilesetBuilder not compiling

### LINKED ISSUE #11: I can't build TilesetBuilder
I followed the instructions "Copy the entire TilesetBuilder directory inside OpenRA's root directory. Build OpenRA if you haven't. Open the TilesetBuilder directory and run make.cmd by double-clicking." but i just got some error and no TilesetBuilder.exe 

Is this tool outdated? 

Here is the entire processes:

```
Security warning
Run only scripts that you trust. While scripts from the internet can be useful,
 this script can potentially harm your computer. Do you want to run
C:\Games\Command And
Conquer\OpenRa\release-20171014\TilesetBuilder-master\make.ps1?
[D] Do not run  [R] Run once  [S] Suspend  [?] Help (default is "D"): r
Microsoft (R) Build Engine version 4.6.1099.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.

Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
Build started 12/17/2017 5:13:26 PM.
Project "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-ma
ster\OpenRA.TilesetBuilder.sln" on node 1 (Rebuild target(s)).
ValidateSolutionConfiguration:
  Building solution configuration "Debug|x86".
Project "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-ma
ster\OpenRA.TilesetBuilder.sln" (1) is building "C:\Games\Command And Conquer\O
penRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.csproj" (2)
on node 1 (Rebuild target(s)).
CoreClean:
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.csprojResolveAssemblyReferen
  ce.cache".
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.FormNew.resources".
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.Properties.Resources.resourc
  es".
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.Surface.resources".
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.FormBuilder.resources".
  Deleting file "C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBu
  ilder-master\obj\x86\Debug\OpenRA.TilesetBuilder.csproj.GenerateResource.Cach
  e".
C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605,5
): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved b
ecause it has an indirect dependency on the assembly "SharpFont, Version=4.0.1.
200, Culture=neutral, PublicKeyToken=48add4c483071cdf" which was built against
the ".NETFramework,Version=v4.5" framework. This is a higher version than the c
urrently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command And
 Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.cs
proj]
C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605,5
): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved b
ecause it has an indirect dependency on the assembly "MaxMind.Db, Version=2.0.0
.0, Culture=neutral, PublicKeyToken=66afa4cc5ae853ac" which was built against t
he ".NETFramework,Version=v4.5" framework. This is a higher version than the cu
rrently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command And
Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.csp
roj]
C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605,5
): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved b
ecause it has an indirect dependency on the assembly "Open.Nat, Version=1.0.0.0
, Culture=neutral, PublicKeyToken=f22a6a4582336c76" which was built against the
 ".NETFramework,Version=v4.5" framework. This is a higher version than the curr
ently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command And Co
nquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.cspro
j]
CoreResGen:
  "C:\Program Files (x86)\Microsoft SDKs\Windows\v7.0A\bin\resgen.exe" /useSour
  cePath /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.N
  ETFramework\v4.0\mscorlib.dll" /r:"C:\Program Files (x86)\Reference Assemblie
  s\Microsoft\Framework\.NETFramework\v4.0\System.Core.dll" /r:"C:\Program File
  s (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.Da
  ta.DataSetExtensions.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Mic
  rosoft\Framework\.NETFramework\v4.0\System.Data.dll" /r:"C:\Program Files (x8
  6)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.dll" /r
  :"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramewo
  rk\v4.0\System.Drawing.dll" /r:"C:\Program Files (x86)\Reference Assemblies\M
  icrosoft\Framework\.NETFramework\v4.0\System.Windows.Forms.dll" /r:"C:\Progra
  m Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\Sys
  tem.Xml.dll" /r:"C:\Program Files (x86)\Reference Assemblies\Microsoft\Framew
  ork\.NETFramework\v4.0\System.Xml.Linq.dll" /compile frmNew.resx,obj\x86\Debu
  g\OpenRA.TilesetBuilder.FormNew.resources Properties\Resources.resx,obj\x86\D
  ebug\OpenRA.TilesetBuilder.Properties.Resources.resources Surface.resx,obj\x8
  6\Debug\OpenRA.TilesetBuilder.Surface.resources FormBuilder.resx,obj\x86\Debu
  g\OpenRA.TilesetBuilder.FormBuilder.resources
  Processing resource file "frmNew.resx" into "obj\x86\Debug\OpenRA.TilesetBuil
  der.FormNew.resources".
  Processing resource file "Properties\Resources.resx" into "obj\x86\Debug\Open
  RA.TilesetBuilder.Properties.Resources.resources".
  Processing resource file "Surface.resx" into "obj\x86\Debug\OpenRA.TilesetBui
  lder.Surface.resources".
  Processing resource file "FormBuilder.resx" into "obj\x86\Debug\OpenRA.Tilese
  tBuilder.FormBuilder.resources".
GenerateTargetFrameworkMonikerAttribute:
Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output fi
les are up-to-date with respect to the input files.
CoreCompile:
  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Csc.exe /noconfig /unsafe+ /n
  owarn:1701,1702 /nostdlib+ /platform:x86 /errorreport:prompt /define:DEBUG;TR
  ACE /highentropyva- /reference:"C:\Program Files (x86)\Reference Assemblies\M
  icrosoft\Framework\.NETFramework\v4.0\mscorlib.dll" /reference:"C:\Program Fi
  les (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.
  Core.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft\F
  ramework\.NETFramework\v4.0\System.Data.DataSetExtensions.dll" /reference:"C:
  \Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v
  4.0\System.Data.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\
  Microsoft\Framework\.NETFramework\v4.0\System.dll" /reference:"C:\Program Fil
  es (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\System.D
  rawing.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsoft
  \Framework\.NETFramework\v4.0\System.Windows.Forms.dll" /reference:"C:\Progra
  m Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.0\Sys
  tem.Xml.dll" /reference:"C:\Program Files (x86)\Reference Assemblies\Microsof
  t\Framework\.NETFramework\v4.0\System.Xml.Linq.dll" /debug+ /debug:full /file
  align:512 /out:obj\x86\Debug\OpenRA.TilesetBuilder.exe /resource:obj\x86\Debu
  g\OpenRA.TilesetBuilder.FormNew.resources /resource:obj\x86\Debug\OpenRA.Tile
  setBuilder.Properties.Resources.resources /resource:obj\x86\Debug\OpenRA.Tile
  setBuilder.Surface.resources /resource:obj\x86\Debug\OpenRA.TilesetBuilder.Fo
  rmBuilder.resources /target:winexe /utf8output /win32icon:"tilesetbuilder_ico
  n copy.ico" Program.cs Properties\Resources.Designer.cs Properties\Settings.D
  esigner.cs Surface.cs Template.cs FormBuilder.cs FormBuilder.Designer.cs Form
  New.cs FormNew.Designer.cs "C:\Users\ELoyros\AppData\Local\Temp\.NETFramework
  ,Version=v4.0.AssemblyAttributes.cs"
FormBuilder.cs(19,14): error CS0234: The type or namespace name 'Graphics' does
 not exist in the namespace 'OpenRA' (are you missing an assembly reference?) [
C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Open
RA.TilesetBuilder.csproj]
Template.cs(8,21): error CS0246: The type or namespace name 'int2' could not be
 found (are you missing a using directive or an assembly reference?) [C:\Games\
Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.Tilese
tBuilder.csproj]
FormBuilder.cs(27,3): error CS0246: The type or namespace name 'TerrainTypeInfo
' could not be found (are you missing a using directive or an assembly referenc
e?) [C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master
\OpenRA.TilesetBuilder.csproj]
Done Building Project "C:\Games\Command And Conquer\OpenRa\release-20171014\Til
esetBuilder-master\OpenRA.TilesetBuilder.csproj" (Rebuild target(s)) -- FAILED.

Done Building Project "C:\Games\Command And Conquer\OpenRa\release-20171014\Til
esetBuilder-master\OpenRA.TilesetBuilder.sln" (Rebuild target(s)) -- FAILED.


Build FAILED.

"C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Ope
nRA.TilesetBuilder.sln" (Rebuild target) (1) ->
"C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Ope
nRA.TilesetBuilder.csproj" (Rebuild target) (2) ->
(ResolveAssemblyReferences target) ->
  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605
,5): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved
 because it has an indirect dependency on the assembly "SharpFont, Version=4.0.
1.200, Culture=neutral, PublicKeyToken=48add4c483071cdf" which was built agains
t the ".NETFramework,Version=v4.5" framework. This is a higher version than the
 currently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command A
nd Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.
csproj]
  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605
,5): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved
 because it has an indirect dependency on the assembly "MaxMind.Db, Version=2.0
.0.0, Culture=neutral, PublicKeyToken=66afa4cc5ae853ac" which was built against
 the ".NETFramework,Version=v4.5" framework. This is a higher version than the
currently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command An
d Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.c
sproj]
  C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Microsoft.Common.targets(1605
,5): warning MSB3275: The primary reference "OpenRA.Game" could not be resolved
 because it has an indirect dependency on the assembly "Open.Nat, Version=1.0.0
.0, Culture=neutral, PublicKeyToken=f22a6a4582336c76" which was built against t
he ".NETFramework,Version=v4.5" framework. This is a higher version than the cu
rrently targeted framework ".NETFramework,Version=v4.0". [C:\Games\Command And
Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.TilesetBuilder.csp
roj]


"C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Ope
nRA.TilesetBuilder.sln" (Rebuild target) (1) ->
"C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Ope
nRA.TilesetBuilder.csproj" (Rebuild target) (2) ->
(CoreCompile target) ->
  FormBuilder.cs(19,14): error CS0234: The type or namespace name 'Graphics' do
es not exist in the namespace 'OpenRA' (are you missing an assembly reference?)
 [C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\Op
enRA.TilesetBuilder.csproj]
  Template.cs(8,21): error CS0246: The type or namespace name 'int2' could not
be found (are you missing a using directive or an assembly reference?) [C:\Game
s\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-master\OpenRA.Tile
setBuilder.csproj]
  FormBuilder.cs(27,3): error CS0246: The type or namespace name 'TerrainTypeIn
fo' could not be found (are you missing a using directive or an assembly refere
nce?) [C:\Games\Command And Conquer\OpenRa\release-20171014\TilesetBuilder-mast
er\OpenRA.TilesetBuilder.csproj]

    3 Warning(s)
    3 Error(s)

Time Elapsed 00:00:00.53

```

### PR DESCRIPTION
**Pull request by abcdefg30**
Adds OpenRA as submodule so we can stay in control of the OpenRA version used to build this project.

Fixes #11.

### CODE REVIEW

**Review by Mailaender — COMMENTED** [review]
The `copy.cmd` promised in the commit message seems to be missing.

File: defaults.yaml [hunk 1]
```diff
@@ -3,49 +3,49 @@ Terrain:
 		Type: Sand
 		AcceptsSmudgeType: SandCrater
 		IsWater: False
```
Comment (reaperrr) [inline defaults.yaml]: Is the TilesetBuilder targeting latest bleed? Then the IsWater entries should be removed now.

Comment (abcdefg30) [reply defaults.yaml]: No, I just revived an old branch.

**Comment by abcdefg30** [comment]
> The `copy.cmd` promised in the commit message seems to be missing.

Ah, ops. I removed that one as it wasn't needed without fixing the commit message.

**Comment by KOYK** [comment]
Is this ready to be merged?

**Comment by abcdefg30** [comment]
I updated this once again and @KOYK confirmed it working on a quick test.

**Comment by KOYK** [comment]
Works perfectly!

**Pull request merged**